x-pack/plugins/enterprise_search/server/routes/app_search/app_search_request_handler.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import fetch from 'node-fetch';
+import querystring from 'querystring';
+import { Logger } from 'src/core/server';
+import { ConfigType } from '../../index';
+
+interface IAppSearchRequestParams {
+  config: ConfigType;
+  log: Logger;
+  path: string;
+  hasValidData: (body) => boolean;
+}
+
+/**
+ * This function creates a handler, in a standard way, for handling App Search
+ * API requests.
+ *
+ * This handler assumes that it will essentially just proxy the App Search API
+ * request, so the request body and request parameters, and body are simply
+ * passed through.
+ */
+export function createAppSearchRequestHandler({
+  config,
+  log,
+  path,
+  hasValidData,
+}: IAppSearchRequestParams) {
+  return async (_context, request, response) => {
+    try {
+      const enterpriseSearchUrl = config.host as string;
+      const params = querystring.stringify(request.query);
+      const url = `${encodeURI(enterpriseSearchUrl)}${path}?${params}`;
+
+      const apiResponse = await fetch(url, {
+        headers: { Authorization: request.headers.authorization as string },
+      });
+
+      const body = await apiResponse.json();
+
+      if (hasValidData(body)) {
```

Comment:
We might want to make `hasValidData` an optional function - it's possible some endpoints may want to just hit the Enterprise Search API without any validation whatsoever, in which case we could maybe do something like:

```suggestion
      if (hasValidData === undefined || hasValidData(body)) {
```

Sorta similar, it's possible future routes will want to pass more dependencies/custom args at some point... 🤔 Maybe we should set up the last arg as an `optionalDependencies` obj? Or am I thinking too far ahead / solving for a problem that doesn't exist yet?